src/main/java/com/example/GymProject/security/CustomAuthFilter.java

```diff
@@ -0,0 +1,51 @@
+package com.example.GymProject.security;
+
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
+import lombok.SneakyThrows;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.context.annotation.Lazy;
+import org.springframework.security.authentication.AuthenticationManager;
+import org.springframework.security.authentication.BadCredentialsException;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.web.authentication.AbstractAuthenticationProcessingFilter;
+import org.springframework.security.web.util.matcher.AntPathRequestMatcher;
+import org.springframework.stereotype.Component;
+
+@Slf4j
+@Component
+public class CustomAuthFilter extends AbstractAuthenticationProcessingFilter {
+    private final LoginAttemptService loginAttemptService;
+
+    public CustomAuthFilter(LoginAttemptService loginAttemptService,
+                            @Lazy AuthenticationManager authenticationManager,
+                            CustomAuthenticationSuccessHandler customAuthenticationSuccessHandler,
+                            CustomAuthenticationFailureHandler customAuthenticationFailureHandler) {
+        super(new AntPathRequestMatcher("/api/login", "POST"), authenticationManager);
+        this.loginAttemptService = loginAttemptService;
+        setAuthenticationSuccessHandler(customAuthenticationSuccessHandler);
+        setAuthenticationFailureHandler(customAuthenticationFailureHandler);
+    }
+
+    @Override
+    @SneakyThrows
```

Comment:
Please avoid this annotation as much as possible.

Reply:
Better to have all things in signature. It's really hard to understand such a code.